.github/workflows/build-deploy.yml

```
@@ -26,74 +30,147 @@ jobs:
         run: echo "::set-output name=id::$(uuidgen)"
     outputs:
       unique_id: ${{ steps.unique_id.outputs.id }}
-  env_name:
+      
+  configure:
+    name: Configure variables
     runs-on: ubuntu-latest
+    outputs:
+      ENVIRONMENT: ${{ steps.configure.outputs.ENVIRONMENT }}
+      PRERELEASE: ${{ steps.configure.outputs.PRERELEASE }}
+      IMAGE_TAG: ${{ steps.configure.outputs.IMAGE_TAG }}
+      TAG_DRY_RUN: ${{ steps.configure.outputs.TAG_DRY_RUN }}
+      PUSH_DOCKER: ${{ steps.configure.outputs.PUSH_DOCKER }}
     steps:
-      - name: Get GHA environment name
-        id: env_name
+      - uses: styfle/cancel-workflow-action@0.11.0
+        with:
+          access_token: ${{ github.token }}
+      - id: configure
         run: |
-          echo "Running on branch ${{ github.ref }}"
-          if [ "${{ github.ref }}" = "refs/heads/develop" ]; then
-            echo "::set-output name=env_name::dev"
-          elif [ "${{ github.ref }}" = "refs/heads/master" ]; then
-            echo "::set-output name=env_name::staging"
-          elif [ "${{ github.ref }}" = "refs/heads/release" ]; then
-            echo "::set-output name=env_name::prod"
-          fi
-      - name: Use specified environment secrets from previous step
-        run: echo "Environment secrets set name ${{ steps.env_name.outputs.env_name }}"
+              echo "Running on branch ${{ github.ref }}"
+              if [ "${{ github.ref }}" = "refs/heads/develop" ]; then
+                echo "ENVIRONMENT=dev" >> $GITHUB_OUTPUT
+                echo "PRERELEASE=true" >> $GITHUB_OUTPUT
+                echo "IMAGE_TAG=$(uuidgen)" >> $GITHUB_OUTPUT
+                echo "TAG_DRY_RUN=true" >> $GITHUB_OUTPUT
+                echo "PUSH_DOCKER=false" >> $GITHUB_OUTPUT
```

Why `false` here? I think we want to push for develop as well, we're just not adding the semvar tag, right?
```suggestion
                echo "PUSH_DOCKER=true" >> $GITHUB_OUTPUT
```